Frontend/Report/frontend-report-first-half-of-november-2024.md

```diff
@@ -0,0 +1,99 @@
+---
+tags:
+  - frontend
+  - market-report
+title: Frontend Report November 2024
+short_title: "Nov 2024"
+description: A comprehensive update on key advancements in frontend development for November 2024, highlighting React 19's full-stack capabilities, XState for state management, Shopify's React Native migration, Next.js 15 enhancements, and the impact of container queries on responsive design.
+date: 2024-11-20
+authors:
+  - hthai2201
+---
+
+A comprehensive November 2024 update on key advancements in frontend development. This report covers the latest trends, tools, and techniques in the frontend ecosystem.
+
+## React
+
+### [React 19’s Full-Stack Evolution: Unifying Server and Client](https://bobaekang.com/blog/two-ways-to-the-two-reacts/)
+
+React 19 introduces full-stack capabilities, enabling components to run on both server and client. Frameworks like Next.js App Router focus on server-driven workflows with caching and SSR, while TanStack Start emphasizes client-side flexibility with server-side primitives. This evolution enhances React's performance and development efficiency.
```

Comment:
I'd put the focus on how TanStack Start is challenging NextJS. It seems to be the focus of this article